src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -544,6 +568,132 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
+{
+	List *citusTableIdList = CitusTableTypeIdList(ANY_CITUS_TABLE_TYPE);
+	Oid citusTableId = InvalidOid;
+	foreach_oid(citusTableId, citusTableIdList)
+	{
+		List *attnumList = NIL;
+		List *dependentSequenceList = NIL;
+		GetDependentSequencesWithRelation(citusTableId, &attnumList,
+										  &dependentSequenceList, 0);
+		ListCell *attnumCell = NULL;
+		ListCell *dependentSequenceCell = NULL;
+		forboth(attnumCell, attnumList, dependentSequenceCell,
+				dependentSequenceList)
+		{
+			AttrNumber currentAttnum = lfirst_int(attnumCell);
+			Oid currentSeqOid = lfirst_oid(dependentSequenceCell);
+
+			/*
+			 * If another distributed table is using the same sequence
+			 * in one of its column defaults, make sure the types of the
+			 * columns match
+			 */
+			if (currentSeqOid == seqOid)
+			{
+				Oid currentSeqTypId = GetAttributeTypeOid(citusTableId,
+														  currentAttnum);
+				if (seqTypId != currentSeqTypId)
+				{
+					char *sequenceName = generate_qualified_relation_name(
+						seqOid);
+					char *citusTableName =
+						generate_qualified_relation_name(citusTableId);
+					ereport(ERROR, (errmsg(
+										"The sequence %s is already used for a different"
+										" type in column %d of the table %s",
+										sequenceName, currentAttnum,
+										citusTableName)));
+				}
+			}
+		}
+	}
+}
+
+
+/*
+ * AlterSequenceType alters the given sequence's type to the given type.
+ */
+void
+AlterSequenceType(Oid seqOid, Oid typeOid)
+{
+	Form_pg_sequence sequenceData = pg_get_sequencedef(seqOid);
+	Oid currentSequenceTypeOid = sequenceData->seqtypid;
+	if (currentSequenceTypeOid != typeOid)
+	{
+		AlterSeqStmt *alterSequenceStatement = makeNode(AlterSeqStmt);
+		char *seqNamespace = get_namespace_name(get_rel_namespace(seqOid));
+		char *seqName = get_rel_name(seqOid);
+		alterSequenceStatement->sequence = makeRangeVar(seqNamespace, seqName, -1);
+		Node *asTypeNode = (Node *) makeTypeNameFromOid(typeOid, -1);
+		SetDefElemArg(alterSequenceStatement, "as", asTypeNode);
+		ParseState *pstate = make_parsestate(NULL);
+		AlterSequence(pstate, alterSequenceStatement);
+	}
+}
+
+
+/*
+ * EnsureSequenceDependenciesAndMarkDist ensures dependencies
+ * for the given sequence exist on all nodes and marks the sequence
+ * as distributed.
+ */
+void
+EnsureSequenceDependenciesAndMarkDist(Oid sequenceOid)
+{
+	/* get sequence address */
+	ObjectAddress sequenceAddress = { 0 };
+	ObjectAddressSet(sequenceAddress, RelationRelationId, sequenceOid);
+	EnsureDependenciesExistOnAllNodes(&sequenceAddress);
+	MarkObjectDistributed(&sequenceAddress);
+}
+
+
+/*
+ * HandleSequencesTypes first ensures that the type of the column in which the sequence is
+ * used as default is supported for each sequence in input dependentSequenceList, and then
+ * alters the sequence type if not the same with the column type.
+ */
+void
+HandleSequencesTypes(Oid relationId, List *dependentSequenceList, List *attnumList)
```

Comment:
Wondering if this can be more descriptive, e.g.: `EnsureDistributedSequencesHaveOneType`